src/resource/resource-opening-hours/ResourceOpeningHours.tsx

```diff
@@ -112,10 +107,55 @@ export default function ResourceOpeningHours({
     []
   );
 
+export default function ResourceOpeningHours({
+  resourceId,
+}: {
+  resourceId: number;
+}): JSX.Element | null {
+  const [error, setError] = useState<Error | undefined>(undefined);
+  const [[defaultPeriods, exceptionPeriods], setDividedDatePeriods] = useState<
+    DatePeriod[][]
+  >([[], []]);
+  const fetchDatePeriods = async (id: number): Promise<void> => {
```

Comment:
When re-render of this component occurs, are all of these functions created again? If that's the case Im wondering if it would be wiser to define them outside of this component?

Reply:
Yes, it might be. It probably has to take a callback as parameter 🤔

Reply:
Probably but I'm not sure does it cause perf issues?